Client/Language.cs

```diff
@@ -0,0 +1,257 @@
+
+        // Craft Library
+        public static string craftLibrary = "Craft Library";
+        public static string plrCraftLibrary = "{0}'s Craft Library";
+        public static string uploadedCraftMsg = "Craft uploaded!";
+        public static string uploadingCraftMsg = "Uploading craft...";
+        public static string savedCraftMsg = "Craft {0} saved!";
+
+        // Screenshot Library
+        public static string screenshots = "Screenshots";
+        public static string sharedScreenshotMsg = "{0} shared a screenshot";
+        public static string screenshotUploadedMsg = "Screenshot uploaded!";
+        public static string uploadingScreenshotMsg = "Uploading screenshot...";
+        public static string downloadingScreenshotMsg = "Downloading screenshot...";
+
+        // Dates
+        public static string singularYear = "year";
+        public static string pluralYear = "years";
+        public static string singularMonth = "month";
+        public static string pluralMonth = "months";
+        public static string singularWeek = "week";
+        public static string pluralWeek = "weeks";
+        public static string singularDay = "day";
+        public static string pluralDay = "days";
+        public static string singularHour = "hour";
+        public static string pluralHour = "hours";
+        public static string singularMinute = "minute";
+        public static string pluralMinute = "minutes";
+        public static string singularSecond = "second";
+        public static string pluralSecond = "seconds";
+        public static string shortYear = "y";
+        public static string shortMonth = "m";
+        public static string shortWeek = "w";
+        public static string shortDay = "d";
+
+    }
+
+    public enum DMPLanguage
+    {
+        ENGLISH,
+        SPANISH,
+        BRAZILIAN,
+        GERMAN
+    }
+
+    public class LanguageWorker
+    {
+        //LanguageWorker
+        private static LanguageWorker singleton = new LanguageWorker();
+        private string dataLocation;
+        private string langFile;
+        private string LANGUAGE_FILE = "english.txt";
+
+        private bool loadedSettings = false;
+
+        private static Dictionary<string, string> languageStrings;
+
+        public static LanguageWorker fetch
+        {
+            get
+            {
+                return singleton;
+            }
+        }
+
+        public LanguageWorker()
+        {
+            dataLocation = Path.Combine(Directory.GetParent(System.Reflection.Assembly.GetExecutingAssembly().Location).Parent.ToString(), "Language");
+            if (!Directory.Exists(dataLocation))
+            {
+                Directory.CreateDirectory(dataLocation);
+            }
+            langFile = Path.Combine(dataLocation, LANGUAGE_FILE);
+            languageStrings = new Dictionary<string, string>();
+        }
+
+        public void LoadLanguage(DMPLanguage lang)
+        {
+            switch (lang)
```

Comment:
We'll want to load/save to a class, but unless we split it up it might get confusing. We'll have to think about the object layout a little - I wouldn't want to hit a dictionary hundreds of times a frame though ;)